pkg/server/server.go

```diff
@@ -402,3 +403,43 @@ func (s *DriverServiceServer) Monitoring(ctx context.Context, req *pb.Monitoring
 	log.Info("Monitoring Routine Done", zap.String("sp", sp.GetUuid()))
 	return &pb.MonitoringResponse{}, nil
 }
+
+func (s *DriverServiceServer) SuspendMonitoring(ctx context.Context, req *pb.MonitoringRequest) (*pb.MonitoringResponse, error) {
+	log := s.log.Named("Monitoring")
+	sp := req.GetServicesProvider()
+	log.Info("Starting Monitoring Routine", zap.String("sp", sp.GetUuid()))
+
+	c, err := one.NewClientFromSP(sp, log)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Error making client: %v", err)
+	}
+
+	for _, group := range req.GetGroups() {
+		for _, inst := range group.GetInstances() {
+			vmid, err := one.GetVMIDFromData(c, inst)
+			if err != nil {
+				log.Warn("Failed to obtain VM ID", zap.String("instance", inst.Uuid), zap.Error(err))
+				return nil, err
+			}
+			_, state, _, _, err := c.StateVM(vmid)
+			if err != nil {
+				log.Warn("Could not get state for VM ID", zap.Int("vmid", vmid))
+				return nil, err
+			}
+			if group.Status != proto.InstanceStatus_SUS && state == "SUSPENDED" {
+				if err := c.ResumeVM(vmid); err != nil {
+					log.Warn("Could not resume VM with VMID", zap.Int("vmid", vmid))
+					return nil, err
+				}
+			}
+
+			if group.Status == proto.InstanceStatus_SUS && state != "SUSPENDED" {
+				if err := c.SuspendVM(vmid); err != nil {
+					log.Warn("Could not suspend VM with VMID", zap.Int("vmid", vmid))
+					return nil, err
```

Comment:
has to be continue (and optionally instance state update), not a return
return will stop the whole function, thus skipping the other groups and instances